admin/includes/classes/observers/class.admin_submenus.php

```diff
@@ -0,0 +1,84 @@
+            foreach($admin_submenus as $table_name => $menus) { // Extract translation data for each table that needs translation.
+                if (empty($table_type[$table_name])) continue;
+                switch ($table_type[$table_name]) { // Different queries are needed because tables have differents fields and unique index keys.
+                    case 1: // Changes configuration table or product_type_layout table columns configuration_title and configuration_description to new language
+                        $query_configuration = $conn->prepare("UPDATE " . $table_name . " SET configuration_title = ?, configuration_description = ? WHERE configuration_key = ?");
+                        foreach($menus as $configuration_key => $translation) {
+                            $query_configuration->bind_param("sss", $translation['title'], $translation['description'], $configuration_key);
+                            $query_configuration->Execute();
+                        }
+                        $query_configuration->close();
+                        break;
+                    case 2: // Changes configuration_group table columns configuration_group_title and configuration_group_description to new language
+                        $query_conf_group = $conn->prepare("UPDATE " . $table_name . " SET configuration_group_title = ?, configuration_group_description = ? WHERE configuration_group_id = ?");
+                        foreach($menus as $configuration_group_id => $translation) {
+                            $query_conf_group->bind_param("ssi", $translation['group_title'], $translation['group_description'], $configuration_group_id);
+                            $query_conf_group->Execute();
+                        }
+                        $query_conf_group->close();
+                        break;
+                    case 3; // Changes product_types table column type_name to new language
+                        $query_types = $conn->prepare("UPDATE " . $table_name . " SET type_name = ? WHERE type_id = ?");
+                        foreach($menus as $type_id => $type_name) {
+                            $query_types->bind_param("si", $type_name, $type_id);
+                            $query_types->Execute();
+                        }
+                        $query_types->close();
+                        break;
+                    case 4; // Changes plugin_control table column description to new language
+                        $query_plugin = $conn->prepare("UPDATE " . $table_name . " SET description = ? WHERE unique_key = ?");
+                        foreach($menus as $type_id => $type_name) {
+                            $query_plugin->bind_param("ss", $type_name, $type_id);
+                            $query_plugin->Execute();
+                        }
+                        $query_plugin->close();
+                        break;
+                }
+            }
+        }
+        $conn->close();
```

Comment:
This wouldn't be necessary if `$conn` comes from the `$db` class object, because the class registers a call to `close` automatically.

Reply:
Done!